advanced/am-pattern-4/values.yaml

```diff
@@ -0,0 +1,293 @@
+      mysql: true
+      # Enable NFS dynamic provisioner for Kubernetes
+      nfsServerProvisioner: true
+
+    # Persisted and shared runtime artifacts for API Manager
+    # See official documentation (from https://apim.docs.wso2.com/en/latest/install-and-setup/setup/reference/common-runtime-and-configuration-artifacts/#persistent-runtime-artifacts)
+    persistentRuntimeArtifacts:
+      # Kubernetes Storage Class to be used to dynamically provision the relevant Persistent Volumes
+      # Only persistent storage solutions supporting ReadWriteMany access mode are applicable (https://kubernetes.io/docs/concepts/storage/persistent-volumes/#access-modes)
+      # Defaults to Kubernetes Storage Class generated using the NFS Server Provisioner (https://github.com/helm/charts/tree/master/stable/nfs-server-provisioner)
+      storageClass: &storage_class "nfs"
+
+      # Define capacities for persistent runtime artifact directories which are shared between instances of the relevant API Manager profile
+      sharedArtifacts:
+        capacity:
+          # For execution plans shared between the Traffic Manager profile instances
+          executionPlans: 20M
+          # For synapse artifacts of APIs shared between the Gateway profile instances
+          synapseConfigs: 50M
+
+      # Persistent runtime artifacts for Apache Solr-based indexing
+      apacheSolrIndexing:
+        # Indicates if persistence of the runtime artifacts for Apache Solr-based indexing is enabled
+        # By default, this is disabled
+        enabled: false
+        # Define capacities for persistent runtime artifact directories
+        capacity:
+          # For persisting the H2 based local Carbon database file
+          carbonDatabase: 50M
+          # For persisting the indexed data
+          solrIndexedData: 50M
+
+    am:
+      # Container image configurations
+      # If a custom image must be used, uncomment 'dockerRegistry' and provide its value
+      # dockerRegistry: ""
+      imageName: "wso2am"
+      imageTag: "3.1.0"
+      # Refer to the Kubernetes documentation on updating images (https://kubernetes.io/docs/concepts/containers/images/#updating-images)
+      imagePullPolicy: Always
+
+      resources:
+        # These are the resource recommendations for running WSO2 API Management product profiles with profile optimization
+        # Resource configurations defined here are applicable for Gateway and Key Manager profile deployments, which use profile optimization
+        requests:
+          memory: "1Gi"
+          cpu: "1000m"
+        limits:
+          memory: "2Gi"
+          cpu: "2000m"
+
+      # API Manager's Gateway specific configurations
+      gateway:
+        internal:
+          # Hostname for Internal Gateway profile
+          hostname: "internal.gateway.am.wso2.com"
+          # Ingress class for internal Gateway
+          ingress:
```

Comment:
@chathuraswso2 you have to fix this along with annotations as in [this](https://github.com/wso2/kubernetes-mi/blob/master/helm/micro-integrator/values.yaml#L62-L70) example. Otherwise, it does not become generic.

Also, @chathuraswso2 you have to fix this in all patterns as suggested, if we are to release this chart. So, please create a separate issue for this and send the fix for all patterns.

Reply:
Not only in the values.yaml, I have change it in here https://github.com/chathuraswso2/kubernetes-apim/blob/678355f55e4758d7c45ac915b95fe88881616bc5/advanced/am-pattern-4/templates/am/gateway/internal/wso2am-pattern-4-am-gateway-internal-ingress.yaml#L21 as well

Reply:
And I only did this in pattern 4 because we only discussed about the pattern 4 in the demo.

Reply:
+1 for @chirangaalwis's suggestion. We will have to parameterize the Ingress annotations as well. Also, please make sure that we do this change for all the other patterns as well. Otherwise, it will be hard to manage, track these changes across all the patterns and no point in doing another release to apply this change for other patterns.

Reply:
@chathuraswso2 I don't see this change in other profiles of API Manager. I can only see the change in Gateway profile Ingresses. Please correct me if I have missed anything.
- https://github.com/chathuraswso2/kubernetes-apim/blob/3.1.x/advanced/am-pattern-1/templates/am/wso2am-pattern-1-am-ingress.yaml
- https://github.com/chathuraswso2/kubernetes-apim/blob/3.1.x/advanced/am-pattern-1/templates/am-analytics/dashboard/wso2am-pattern-1-am-analytics-dashboard-ingress.yaml
- https://github.com/chathuraswso2/kubernetes-apim/blob/3.1.x/advanced/am-pattern-2/templates/am/pub-devportal-tm/wso2am-pattern-2-am-ingress.yaml
- https://github.com/chathuraswso2/kubernetes-apim/blob/3.1.x/advanced/am-pattern-4/templates/am/pub-devportal-tm/wso2am-pattern-4-am-ingress.yaml
- https://github.com/chathuraswso2/kubernetes-apim/blob/3.1.x/advanced/am-pattern-3/templates/am/publisher/wso2am-pattern-3-am-publisher-ingress.yaml
- https://github.com/chathuraswso2/kubernetes-apim/blob/3.1.x/advanced/am-pattern-3/templates/am/devportal/wso2am-pattern-3-am-devportal-ingress.yaml

Please fix this in every profile and please test it properly.

Reply:
@chathuraswso2 please correct me if I have missed anything but I don't see this change in other product profile Ingresses.

- https://github.com/chathuraswso2/kubernetes-apim/blob/3.1.x/advanced/am-pattern-1/templates/am/wso2am-pattern-1-am-ingress.yaml
- https://github.com/chathuraswso2/kubernetes-apim/blob/3.1.x/advanced/am-pattern-1/templates/am-analytics/dashboard/wso2am-pattern-1-am-analytics-dashboard-ingress.yaml
- https://github.com/chathuraswso2/kubernetes-apim/blob/3.1.x/advanced/am-pattern-2/templates/am/pub-devportal-tm/wso2am-pattern-2-am-ingress.yaml
- https://github.com/chathuraswso2/kubernetes-apim/blob/3.1.x/advanced/am-pattern-2/templates/am-analytics/dashboard/wso2am-pattern-2-am-analytics-dashboard-ingress.yaml
- https://github.com/chathuraswso2/kubernetes-apim/blob/3.1.x/advanced/am-pattern-4/templates/am/pub-devportal-tm/wso2am-pattern-4-am-ingress.yaml
- https://github.com/chathuraswso2/kubernetes-apim/blob/3.1.x/advanced/am-pattern-4/templates/am-analytics/dashboard/wso2am-pattern-4-am-analytics-dashboard-ingress.yaml
- https://github.com/chathuraswso2/kubernetes-apim/blob/3.1.x/advanced/am-pattern-3/templates/am/publisher/wso2am-pattern-3-am-publisher-ingress.yaml
- https://github.com/chathuraswso2/kubernetes-apim/blob/3.1.x/advanced/am-pattern-3/templates/am/devportal/wso2am-pattern-3-am-devportal-ingress.yaml
- https://github.com/chathuraswso2/kubernetes-apim/blob/3.1.x/advanced/am-pattern-3/templates/am-analytics/dashboard/wso2am-pattern-3-am-analytics-dashboard-ingress.yaml

Please fix them appropriately and test it properly.

Reply:
Fix in https://github.com/wso2/kubernetes-apim/pull/444/commits/5d99c7a0cc84924f01b31dbbba09b9169c67e61d